File: cpp/warshalls.cpp
#include<iostream>
using namespace std;
int warshalls(int **d,int n)
{
	for (int k = 0; k < n; k++)
	{
		cout << "R(" << (k+1) << ")=" << endl;
		for (int i = 0; i < n; i++)
		{
			for (int j = 0; j < n; j++)
			{
				d[i][j]=d[i][j]|| (d[i][k] && d[k][j]);
				cout << d[i][j] << "\t";
			}
			cout << endl;
		}
		cout << endl;
	}
	return n;
}
int main()
{
	cout << "Enter the number of nodes:";
	int n;
	cin >> n;
	int **d=new int*[n];
	for (int i = 0; i < n; i++)
	{
		d[i]=new int[n];
	}
	cout << "Enter the Matrix:" << endl;
	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < n; j++)
		{
			cin >> d[i][j];
		}
	}
	int r=warshalls(d,n);
	for (int i = 0; i < n; i++)
	{
		delete[] d[i];
	}
	delete[] d;
	
	return 0;
}

File: java/4th semester/Warshalls.java
import java.util.Scanner;
public class Warshalls
{
    void warshalls(int a[][],int n)
    {
        for(int k=0;k<n;k++)
        {
            System.out.print("\nR("+(k+1)+")=");
            for(int i=0;i<n;i++)
            {
                System.out.println();
                for(int j=0;j<n;j++)
                {
                    a[i][j]=(a[i][j]!=0||(a[i][k]!=0&&a[k][j]!=0))?1:0;
                    System.out.print(a[i][j]+"\t");
                }
            }
        }
    }
    public static void main(String[] args)
    {
        Scanner s=new Scanner(System.in);
        Warshalls w=new Warshalls();
        System.out.println("Enter the number of nodes:");
        int n=s.nextInt();
        int a[][]=new int[n][n];
        System.out.println("Enter the Transitive Closure:");
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<n;j++)
            {
                a[i][j]=s.nextInt();
            }
        }
        w.warshalls(a,n);
        s.close();
    }
}
